exec/java-exec/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -464,22 +464,30 @@ SqlNode SqlDropSchema(SqlParserPos pos) :
 
 /**
  * Parse refresh table metadata statement.
- * REFRESH TABLE METADATA tblname
+ * REFRESH TABLE METADATA [COLUMNS ((field1, field2,..) | NONE)] tblname
  */
 SqlNode SqlRefreshMetadata() :
 {
     SqlParserPos pos;
     SqlIdentifier tblName;
-    SqlNodeList fieldList;
+    SqlNodeList fieldList = null;
     SqlNode query;
+    boolean allColumns = true;
```

Comment:
I don't think we need allColumns state here, if fieldList is empty it means we refresh all columns, if not then part of them. Having `SqlNodeList fieldList` is enough.

Reply:
We have three scenarios here:
Refresh metadata for all the columns -  allColumns - true, fieldList - []
Refresh metadata for subset of columns - allColumns -false, fieldList - [c1,c2,..]
Refresh metadata and don't collect column metadata - allColumns - false, fieldList - []
Since we have three scenarios we can't represent all three by using just one variable.